File: packages/auth-client/src/python/aeos_auth_client/agent_identity.py
from __future__ import annotations

import os
import httpx

from .types import AgentIdentityVerification


async def verify_agent_contract(contract_id: str, agent_id: str) -> AgentIdentityVerification:
    auth_service_url = os.environ.get("AUTH_SERVICE_URL")
    if not auth_service_url:
        raise RuntimeError("AUTH_SERVICE_URL not configured")

    async with httpx.AsyncClient() as client:
        response = await client.post(
            f"{auth_service_url}/v1/agent-contracts/{contract_id}/verify",
            json={"agent_id": agent_id},
        )

    if not response.is_success:
        return AgentIdentityVerification(valid=False)

    body = response.json()
    return AgentIdentityVerification(
        valid=body["valid"],
        agent_id=body.get("agent_id"),
        uop_id=body.get("uop_id"),
    )

File: packages/auth-client/src/agent-identity.ts
import type { AgentContractId } from '@aeos/canonical-schema';

export interface AgentIdentityVerification {
  readonly valid: boolean;
  readonly agentId?: string;
  readonly uopId?: string;
}

export async function verifyAgentContract(
  contractId: AgentContractId,
  agentId: string,
): Promise<AgentIdentityVerification> {
  const authServiceUrl = process.env['AUTH_SERVICE_URL'];
  if (!authServiceUrl) {
    throw new Error('AUTH_SERVICE_URL not configured');
  }

  const response = await fetch(`${authServiceUrl}/v1/agent-contracts/${contractId}/verify`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ agent_id: agentId }),
  });

  if (!response.ok) {
    return { valid: false };
  }

  const body = (await response.json()) as {
    valid: boolean;
    agent_id?: string;
    uop_id?: string;
  };
  return {
    valid: body.valid,
    agentId: body.agent_id,
    uopId: body.uop_id,
  };
}
